compendium/LogServiceImpl/src/LogServiceImpl.cpp

```diff
@@ -6,110 +6,193 @@
+	{
+	    auto currLogger = std::atomic_load(&logger);
+	    if (!currLogger)
+	    {
+		return;
+	    }
+	    switch (level)
+	    {
+		case SeverityLevel::LOG_DEBUG:
+		{
+		    currLogger->debug(message, sr);
+		    break;
+		}
+		case SeverityLevel::LOG_INFO:
+		{
+		    currLogger->info(message, sr);
+		    break;
+		}
+		case SeverityLevel::LOG_WARNING:
+		{
+		    currLogger->warn(message, sr);
+		    break;
+		}
+		case SeverityLevel::LOG_ERROR:
+		{
+	    	    currLogger->error(message, sr);
+		    break;	
+		}
+		default:
+		{
+		    currLogger->trace(message, sr);
+		    break;
+		}
+	    }	
+	}
 
-        void
-        LogServiceImpl::Log(ServiceReferenceBase const& sr, SeverityLevel level, std::string const& message)
+	void
+	LogServiceImpl::Log(ServiceReferenceBase const& sr,
+				SeverityLevel level,
+				std::string const& message,
+				const std::exception_ptr ex)
+	{
+
+	    auto currLogger = std::atomic_load(&logger);
+	    if(!currLogger)
+	    {
+		return;
+       	    }
+	    switch (level)
+	    {
+	        case SeverityLevel::LOG_DEBUG:
+		{
+		    currLogger->debug(message, sr, ex);
+		    break;
+		}
+		case SeverityLevel::LOG_INFO:
+		{	    
+		    currLogger->info(message, sr, ex);	    
+		    break;
+		}
+		case SeverityLevel::LOG_WARNING:
+		{
+		    currLogger->warn(message, sr, ex);
+		    break;
+		}
+		case SeverityLevel::LOG_ERROR:
+		{
+		    currLogger->error(message, sr, ex);	    
+		    break;
+		}
+		default:
+		{
+		    currLogger->trace(message, sr, ex);
+		    break;
+		}
+   	     }
+	}
+
+        std::shared_ptr<Logger> 
+            LogServiceImpl::getLogger(std::string const& name) const
         {
-            std::string full_message = message;
-            full_message = message + GetServiceReferenceInfo(sr);
-            LogServiceImpl::Log(level, full_message);
+            std::shared_ptr<LoggerFactory> lf = std::make_shared<LoggerFactoryImpl>();
+            return lf->getLogger(name);
         }
 
-        void
-        LogServiceImpl::Log(ServiceReferenceBase const& sr,
-                            SeverityLevel level,
-                            std::string const& message,
-                            const std::exception_ptr ex)
+	std::shared_ptr<Logger>
+            LogServiceImpl::getLogger(const cppmicroservices::Bundle& bundle, std::string const& name) const
         {
-            std::string full_message = message;
-            full_message = message + GetServiceReferenceInfo(sr) + GetExceptionMessage(ex);
-            LogServiceImpl::Log(level, full_message);
+            std::shared_ptr<LoggerFactory> lf = std::make_shared<LoggerFactoryImpl>();
+            return lf->getLogger(bundle, name);
         }
 
         void
         LogServiceImpl::AddSink(spdlog::sink_ptr& sink)
         {
-            m_Logger->sinks().push_back(sink);
+	    std::shared_ptr<LoggerImpl> logimpl = std::dynamic_pointer_cast<LoggerImpl>(logger);
```

Comment:
does this instance of accessing `logger` need to follow the same pattern of calling `std::atomic_load(&logger);` first before accessing the instance?